doc/release/upcoming_changes/16056.deprecation.rst

```diff
@@ -0,0 +1,15 @@
+Inexact matches for mode, order and searchside are deprecated
+-------------------------------------------------------------
+Inexact and case insensitive matches for mode, order and searchside were
+valid inputs earlier and will give a DeprecationWarning now.
+For example, below are some example usages which are now deprecated and will
+give a DeprecationWarning.
+
+    import numpy as np
+    arr = np.array([[3, 6, 6], [4, 5, 1]])
+    # order: case insensitive match
+    np.ravel_multi_index(arr, (7, 6), order="f")  # should be "F"
```

Comment:
Nit: Maybe use `np.array()`, or `copy` which also have the order parameter and are more prominent.